core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -92,29 +86,23 @@ public NetworkService(Settings settings, List<CustomNameResolver> customNameReso
      * not contain duplicate addresses.
      *
      * @param bindHosts list of hosts to bind to. this may contain special pseudo-hostnames
-     *                  such as _local_ (see the documentation). if it is null, it will be populated
-     *                  based on global default settings.
+     *                  such as _local_ (see the documentation). if it is null, it will fall back to _local_
+     *
      * @return unique set of internet addresses
      */
     public InetAddress[] resolveBindHostAddresses(String bindHosts[]) throws IOException {
-        // first check settings
         if (bindHosts == null || bindHosts.length == 0) {
-            if (GLOBAL_NETWORK_BINDHOST_SETTING.exists(settings) || GLOBAL_NETWORK_HOST_SETTING.exists(settings)) {
-                // if we have settings use them (we have a fallback to GLOBAL_NETWORK_HOST_SETTING inline
-                bindHosts = GLOBAL_NETWORK_BINDHOST_SETTING.get(settings).toArray(Strings.EMPTY_ARRAY);
-            } else {
-                // next check any registered custom resolvers if any
-                if (customNameResolvers != null) {
-                    for (CustomNameResolver customNameResolver : customNameResolvers) {
-                        InetAddress addresses[] = customNameResolver.resolveDefault();
-                        if (addresses != null) {
-                            return addresses;
-                        }
+            // next check any registered custom resolvers if any
+            if (customNameResolvers != null) {
```

Comment:
I don't think this can ever be null (it is initialized as an empty list when filling in Node), so you can add Object.requireNotNull in the ctor and remove this check?